### PR TITLE
feat(subscriptions): Add cancel to productConfig

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- As part of the metadata expansion project, update the product config
  to include the cancellation survey URL.

## This pull request

- Add the product:cancellationSurveyURL to the productConfig

## Issue that this pull request solves

Closes: #12076

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by StaberindeZA** [comment]
@biancadanforth is this all the work that's required for this ticket?

I tried running the `stripe-products-and-plans-to-firestore-documents` script to see if the cancellation survey URL, but had some trouble getting it to work. That said, from what I could see in the debugging, it did look like the URL was included in the metadata.

**Comment by StaberindeZA** [comment]
TLDR; required joi validations are already done in the ProductConfig class.

We'd initially discussed adding the `required()` check to the joi validation as well, but after further investigation, I found that the Product/Plan required validations are added in the ProductConfig/PlanConfig classes.

**Review by IvoJP — APPROVED** [review]
